<?xml version="1.0"?>
<launch>
  <!-- common -->
  <arg name="common_param_path"/>
  <arg name="vehicle_param_file"/>
  <arg name="nearest_search_param_path"/>

  <!-- launch module -->
  <arg name="behavior_velocity_planner_launch_modules"/>

  <!-- module config path -->
  <arg name="behavior_velocity_config_path"/>
  <arg name="behavior_velocity_smoother_type_param_path"/>
  <arg name="behavior_velocity_planner_param_path"/>
  <arg name="behavior_velocity_planner_common_param_path"/>
  <arg name="behavior_velocity_planner_blind_spot_module_param_path"/>
  <arg name="behavior_velocity_planner_crosswalk_module_param_path"/>
  <arg name="behavior_velocity_planner_walkway_module_param_path"/>
  <arg name="behavior_velocity_planner_detection_area_module_param_path"/>
  <arg name="behavior_velocity_planner_intersection_module_param_path"/>
  <arg name="behavior_velocity_planner_stop_line_module_param_path"/>
  <arg name="behavior_velocity_planner_traffic_light_module_param_path"/>
  <arg name="behavior_velocity_planner_virtual_traffic_light_module_param_path"/>
  <arg name="behavior_velocity_planner_occlusion_spot_module_param_path"/>
  <arg name="behavior_velocity_planner_no_stopping_area_module_param_path"/>
  <arg name="behavior_velocity_planner_run_out_module_param_path"/>
  <arg name="behavior_velocity_planner_speed_bump_module_param_path"/>
  <arg name="behavior_velocity_planner_no_drivable_lane_module_param_path"/>
  <!-- <arg name="behavior_velocity_planner_template_module_param_path"/> -->

  <node pkg="autoware_behavior_velocity_planner" exec="behavior_velocity_planner_node" name="behavior_velocity_planner" output="screen">
    <!-- topic remap -->
    <remap from="~/input/path_with_lane_id" to="path_with_lane_id"/>
    <remap from="~/input/vector_map" to="/map/vector_map"/>
    <remap from="~/input/vehicle_odometry" to="/localization/kinematic_state"/>
    <remap from="~/input/accel" to="/localization/acceleration"/>
    <remap from="~/input/dynamic_objects" to="/perception/object_recognition/objects"/>
    <remap from="~/input/no_ground_pointcloud" to="/perception/obstacle_segmentation/pointcloud"/>
    <remap from="~/input/compare_map_filtered_pointcloud" to="compare_map_filtered/pointcloud"/>
    <remap from="~/input/vector_map_inside_area_filtered_pointcloud" to="vector_map_inside_area_filtered/pointcloud"/>
    <remap from="~/input/external_velocity_limit_mps" to="/planning/scenario_planning/max_velocity_default"/>
    <remap from="~/input/traffic_signals" to="/perception/traffic_light_recognition/traffic_signals"/>
    <remap from="~/input/virtual_traffic_light_states" to="/perception/virtual_traffic_light_states"/>
    <remap from="~/input/occupancy_grid" to="/perception/occupancy_grid_map/map"/>
    <remap from="~/output/path" to="path"/>
    <remap from="~/output/stop_reasons" to="/planning/scenario_planning/status/stop_reasons"/>
    <remap from="~/output/infrastructure_commands" to="/planning/scenario_planning/status/infrastructure_commands"/>
    <remap from="~/output/traffic_signal" to="debug/traffic_signal"/>
    <!-- params -->
    <param name="launch_modules" value="$(var behavior_velocity_planner_launch_modules)"/>
    <!-- load config -->
    <param from="$(var common_param_path)"/>
    <param from="$(var vehicle_param_file)"/>
    <param from="$(var nearest_search_param_path)"/>
    <param from="$(var behavior_velocity_smoother_type_param_path)"/>
    <param from="$(var behavior_velocity_planner_param_path)"/>
    <param from="$(var behavior_velocity_planner_common_param_path)"/>
    <param from="$(var behavior_velocity_planner_blind_spot_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_crosswalk_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_walkway_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_detection_area_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_intersection_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_stop_line_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_traffic_light_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_virtual_traffic_light_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_occlusion_spot_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_no_stopping_area_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_run_out_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_speed_bump_module_param_path)"/>
    <param from="$(var behavior_velocity_planner_no_drivable_lane_module_param_path)"/>
    <!-- <param from="$(var behavior_velocity_planner_template_param_path)"/> -->
  </node>
</launch>
